plonk-core/src/lookup/witness_table.rs

```diff
@@ -0,0 +1,172 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::error::Error;
+use crate::lookup::{LookupTable, MultiSet};
+use ark_ff::Field;
+
+/// This witness table contains quieries to a lookup table for lookup gates.
+/// This table can have any arity. (But for the time other parts of the codebase
+/// force 4-arity)
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct WitnessTable<F>
+where
+    F: Field,
+{
+    /// Vector containing one `MulitSet` for each wire
+    pub f: Vec<MultiSet<F>>,
+}
+
+impl<F> WitnessTable<F>
+where
+    F: Field,
+{
+    /// Initialses empty witness table of arity 4
+    pub fn new() -> Self {
+        Self {
+            f: vec![MultiSet::new(); 4],
+        }
+    }
+
+    /// This allows the witness table to be filled directly without
+    /// taking any vaules, or the the results, from the lookup table.
+    /// If the values do no exists in the lookup table, then the proof
+    /// will fail when witness and preprocessed tables are concatenated.
+    pub fn from_wire_values(&mut self, wires: Vec<F>) {
+        assert_eq!(wires.len(), self.f.len());
+        wires
+            .into_iter()
+            .zip(self.f.iter_mut())
+            .for_each(|(val, column)| column.push(val));
+    }
+
+    /// Attempts to look up a value from a lookup table. If successful, all four
+    /// elements are pushed to their respective multisets.
+    pub fn value_from_table(
+        &mut self,
+        lookup_table: &LookupTable<F>,
+        left_wire_val: F,
+        right_wire_val: F,
+        fourth_wire_val: F,
+    ) -> Result<(), Error> {
+        // This function is specific for 4-arity
```

Comment:
I think this may be useful for contributors that may changes to the library but probably not relevant for users, so I think it is fine as a code comment.